src/nom.rs

```diff
@@ -0,0 +1,255 @@
+        }
+    }
+}
+
+impl Nom for MozjsCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Mozjs::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions += 1;
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for JavascriptCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Javascript::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions += 1;
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for TypescriptCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Typescript::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions += 1;
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for TsxCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Tsx::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions += 1;
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for RustCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Rust::*;
+
+        match node.kind_id().into() {
+            FunctionItem => {
+                stats.functions += 1;
+            }
+            ClosureExpression => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for CppCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Cpp::*;
+
+        match node.kind_id().into() {
+            FunctionDefinition
+            | FunctionDefinition2
+            | FunctionDefinition3
+            | FunctionDefinition4
+            | FunctionDefinitionRepeat1 => {
+                stats.functions += 1;
+            }
+            LambdaExpression => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for PreprocCode {}
+impl Nom for CcommentCode {}
+impl Nom for CSharpCode {}
+impl Nom for JavaCode {}
+impl Nom for GoCode {}
+impl Nom for CssCode {}
+impl Nom for HtmlCode {}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use super::*;
+
+    #[test]
+    fn test_nom_python() {
+        let sample =
```

Comment:
Too much redundant code in these tests.
You could probably add a function in tools.rs to get a parser from a string and a filename (the language would be guessed with the file extension).